src/Core/Framework/Adapter/Twig/TwigEnvironment.php

```diff
@@ -34,14 +35,48 @@ public function compile(Node $node): string
         }
 
         $source = $this->compiler->compile($node)->getSource();
+        $source = $this->addMacroResultCall($source);
 
         $replaces = [
             'CoreExtension::getAttribute(' => 'SwTwigFunction::getAttribute(',
-            'CoreExtension::callMacro(' => 'SwTwigFunction::callMacro(',
             'twig_escape_filter(' => 'SwTwigFunction::escapeFilter(',
             'use Twig\Environment;' => "use Twig\Environment;\nuse Shopware\Core\Framework\Adapter\Twig\SwTwigFunction;",
         ];
 
         return str_replace(array_keys($replaces), array_values($replaces), $source);
     }
+
+    private function addMacroResultCall(string $source): string
+    {
+        if (
+            str_contains($source, 'getTemplateForMacro')
```

Comment:
wow this if is a mess 🙈 

I'm wondering do we still need the yield here at all now that we now use the macro result directly in the get attribute?

Reply:
Sure, it is a mess, and the replacements are a mess, too. 
But somehow, we need to call our result, and that result can be different depending on the macro.

The get attribute is not always executed or added to the different macro PHP templates.

Reply:
Ok can we add a comment to explain it? 

What I understand are the first two parts of the if clause -> we need to adjust macro calls that use our custom return keyword right?